chia/consensus/get_block_challenge.py

```diff
@@ -101,3 +101,30 @@ def get_block_challenge(
                 curr = blocks.block_record(curr.prev_hash)
             challenge = reversed_challenge_hashes[challenges_to_look_for - 1]
     return challenge
+
+
+def prev_tx_block(
+    blocks: BlockRecordsProtocol,
+    prev_b: Optional[Union[BlockRecord, FullBlock, HeaderBlock]],
+) -> uint32:
+    # todo add check to make sure we dont return tx block from same sp as block we are validating
+    if prev_b is None:
+        return uint32(0)
+    if isinstance(prev_b, BlockRecord):
+        if prev_b.prev_transaction_block_hash is not None:
+            return prev_b.height
+        else:
+            curr = prev_b
+    elif isinstance(prev_b, FullBlock):
+        if prev_b.foliage_transaction_block is not None:
+            return prev_b.height
+        else:
+            curr = blocks.block_record(prev_b.header_hash)
+    elif isinstance(prev_b, HeaderBlock):
+        if prev_b.foliage_transaction_block is not None:
+            return prev_b.height
+        else:
+            curr = blocks.block_record(prev_b.header_hash)
```

Comment:
maybe
```
elif:  # pragma: no cover
   assert False, "invalid prev_b passed to prev_tx_block()"
```

I don't really have a strong opinion, it just makes it look a bit scary to leave this case just falling off the cliff rather than failing loudly

Reply:
i trust the linter i think also everything blows up if none of the types match since curr would be None (also this would fail the pre-commit as well)